### PR TITLE
feature(826391): Server side Excel export support for Pivot table is implemented.

### PR DESCRIPTION
**Pull request by Mohanraj-Elumalai**
### Feature description
Provide the support for Server side Excel export for Pivot table

### Requirement and specification document.
 https://syncfusion.sharepoint.com/:w:/r/sites/EJ2-ClientSide/_layouts/15/Doc.aspx?sourcedoc=%7B266BA6A7-B17D-4C34-82BB-D8A7CCEFCBE4%7D&file=EJ2-56334-Support-to-provide-custom-sorting-in-Pivot-Table.docx&action=default&mobileredirect=true - Connect to preview 

### API Review task
New API are introduced for this server side excel export feature support.

**Code Snippets:**
code-snippet-1
code-snippet-2

### Output screenshots
excel-1
excel-2
csv-1

### Feature matrix document

Feature matrix document updated against this feature and committed in this [common location] (https://syncfusion.sharepoint.com/sites/EJ2ProductOwners/Shared%20Documents/Forms/AllItems.aspx?viewid=ae81c682%2D3d0f%2D462a%2Db8ec%2D7358748d489d&id=%2Fsites%2FEJ2ProductOwners%2FShared%20Documents%2FGeneral%2FFeature%20Matrix%20%2D%20Documents) . 
* [ ] Yes
* [ ] NO
* [X] NA

Provide the details about the areas or combinations which have been tested against this code changes.
* [X]  Tested against feature matrix.
1.Tested the excel and csv export
2. Checked with multiple fields
3. Checked the conditional formatting
4. Also checked the formats for values
        
### Automation details
Mark 'Is Automated' field as (Yes, Manual, Not Applicable) in corresponding JIRA task once the feature is automated. 
* [ ] BUnit, share corresponding MR.
* [ ] E2E or Manual Automation using tester - Make sure all items are automated with priority before release which can be tracked in [automation dashboard](https://syncfusion.atlassian.net/secure/Dashboard.jspa?selectPageId=43396).

### Test bed sample location
Provide the test bed sample location where the code reviewers can review the new feature’s behaviors. 

### Feature completion checklist

UX changes got approval if UI is added or modified?
* [ ] Yes. Provide Task link.
* [ ] NO
* [X] NA

Drafted UG for this feature?
* [ ] Yes, provide MR link.
* [X] NO, provide task link.
* [ ] NA

 Content team reviewed the UI content changes.
* [ ] Yes. Provide MR or Task link.
* [x] NO
* [ ] NA

 Is this the common feature which needs to be addressed in the same component or on other components in our platform? 
* [ ]  Yes. If we need to check in other components, tag "needs-attention-coreteam" 
* [X]  NO

Is there any existing behavior change due to this code change?
* [ ]  Yes. If So, add `breaking-change` label.
* [X]  NO

### Blazor Checklist
Confirm whether the ensured feature is in both Blazor Server and WASM.
* [X] Yes
* [ ] NO
* [ ] NA

Do the code changes cause any memory leak and performance issue?
* [ ] Yes
* [X] NO

## Reviewer Checklist
* [ ]  Reviewed feature matrix document modifications and reviewed developer testing report against the feature matrix reviewed.
* [ ]  Testbed sample ensured.
* [ ]  Coverage report checked.
* [ ]  Ensured the code changes meet the coding standard.
* [ ]  Confirming script changes made in this merge request can't be achieved in C#. 

Created task in Blazor to synchronize if the feature is not available and applicable.
* [ ] NA
* [ ] Yes, task link.
* [ ] NO, share reason.

### CODE REVIEW

**Review by ManiRam94 — APPROVED** [review]

File: PivotController/PivotController/Controllers/PivotController.cs [hunk 1]
```diff
@@ -36,6 +37,34 @@ public async Task<object> Post([FromBody]object args)
             {
                 return await GetRawData(param);
             }
+            else if (param.Action == "onExcelExport" || param.Action == "onCsvExport")
+            {
+                if (param.InternalProperties.EnableVirtualization && param.InternalProperties.ExportAllPages)
+                {
+                    EngineProperties engine = await GetEngine(param);
+                    engine = await PivotEngine.PerformAction(engine, param);
```
Comment (SasthaPrathap) [inline PivotController/PivotController/Controllers/PivotController.cs:49]: provide if condition for this line alone.

**Review by SasthaPrathap — CHANGES_REQUESTED** [review]
Please check and clear the review comments.

File: PivotController/PivotController/Controllers/PivotController.cs (hunk 1 above)
Comment (Mohanraj-Elumalai) [reply PivotController/PivotController/Controllers/PivotController.cs:49]: Resolved

File: PivotController/PivotController/Controllers/PivotController.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ public class PivotController : Controller
         private IWebHostEnvironment _hostingEnvironment;
         private bool isRendered;
         private PivotEngine<DataSource.PivotViewData> PivotEngine = new PivotEngine<DataSource.PivotViewData>();
+        private ExcelExport excelExport = new ExcelExport();
```
Comment (SasthaPrathap) [inline PivotController/PivotController/Controllers/PivotController.cs:20]: Add export events for document customization.

**Review by SasthaPrathap — CHANGES_REQUESTED** [review]
Please check and clear the review comment.

File: PivotController/PivotController/Controllers/PivotController.cs (hunk 2 above)
Comment (Mohanraj-Elumalai) [reply PivotController/PivotController/Controllers/PivotController.cs:20]: Added the events in sample

**Review by SasthaPrathap — APPROVED** [review]

**Review by SasthaPrathap — APPROVED** [review]

**Pull request closed without merging**